test/ShapeshiftHallOfFame.test.ts

```diff
@@ -0,0 +1,172 @@
+import * as constants from "./constants";
+
+describe("ShapeshiftHallOfFame", async function () {
+  let accounts: SignerWithAddress[];
+  let nftContract: ShapeshiftHallOfFame;
+
+  beforeEach(async () => {
+    accounts = await ethers.getSigners();
+
+    const nftContractDeployment = await ethers.getContractFactory(
+      "ShapeshiftHallOfFame"
+    );
+    nftContract = await nftContractDeployment.deploy(
+      constants.TIMELOCK_DURATION
+    );
+  });
+
+  describe("minting", function () {
+    it("Should mint the token", async function () {
+      const recipient1 = accounts[0].address;
+
+      // Zero initial balance
+      await expect(await nftContract.balanceOf(recipient1)).eq(0);
+
+      await nftContract.safeMint(recipient1, constants.TEST_URI);
+
+      // Expect to have balance
+      await expect(await nftContract.balanceOf(recipient1)).eq(1);
+
+      // Confirm URI
+      await expect(await nftContract.tokenURI(0)).eq(
+        "ipfs://" + constants.TEST_URI
+      );
+    });
+  });
+
+  describe("edit & transfer", function () {
+    it("Should be able to update the token URI", async function () {
+      const recipient1 = accounts[0].address;
+
+      await nftContract.safeMint(recipient1, constants.TEST_URI);
+
+      // Confirm URI
+      await expect(await nftContract.tokenURI(0)).eq(
+        "ipfs://" + constants.TEST_URI
+      );
+
+      // Wait for the timelock to be lifted
+      await network.provider.send("evm_increaseTime", [
+        constants.TIMELOCK_DURATION,
+      ]);
+
+      await nftContract.setTokenURI(0, constants.TEST_URI_V2);
+
+      // Confirm updated URI
+      await expect(await nftContract.tokenURI(0)).eq(
+        "ipfs://" + constants.TEST_URI_V2
+      );
+    });
+
+    it("Should be able to transfer the token", async function () {
+      const recipient1 = accounts[0].address;
+      const recipient2 = accounts[1].address;
+
+      await nftContract.safeMint(recipient1, constants.TEST_URI);
+
+      // Confirm URI
+      await expect(await nftContract.tokenURI(0)).eq(
+        "ipfs://" + constants.TEST_URI
+      );
+
+      // Wait for the timelock to be lifted
+      await network.provider.send("evm_increaseTime", [
+        constants.TIMELOCK_DURATION,
+      ]);
+
+      // Zero initial balance for recipient2
+      await expect(await nftContract.balanceOf(recipient2)).eq(0);
+
+      await nftContract.approve(recipient2, 0);
+      await nftContract["safeTransferFrom(address,address,uint256)"](
+        recipient1,
+        recipient2,
+        0
+      );
+
+      // Expect recipient2 to have balance
+      await expect(await nftContract.balanceOf(recipient2)).eq(1);
+
+      // Expect recipient1 to have zero balance
+      await expect(await nftContract.balanceOf(recipient1)).eq(0);
+    });
+
+    it("Should be able to change the timelock duration", async function () {
+      const recipient1 = accounts[0].address;
```

Comment:
same here: may be good to add a test to show an update timelock duration from a non owner address fails